pyshield/stencils/shallow_convection/state.py

```diff
@@ -0,0 +1,243 @@
+from dataclasses import dataclass, field, fields
+from typing import Any, Dict, Mapping
+
+import xarray as xr
+
+import ndsl.dsl.gt4py_utils as gt_utils
+from ndsl import GridSizer, Quantity, QuantityFactory
+from ndsl.constants import X_DIM, Y_DIM, Z_DIM
+from ndsl.dsl.typing import Float, Int
+from pyshield.stencils.shallow_convection._config import SC_TRACER_DIM
+
+
+@dataclass()
+class SAMFShalConvState:
```

Comment:
Could be wrapped into the new `ndsl.State`. Can also be a future refactor

Reply:
With `NDSLRuntime` and `Locals`, you have more work for optimziation ahead anyway